pmtiles.cpp

```diff
@@ -0,0 +1,418 @@
+#include <stdio.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <zlib.h>
+#include "pmtiles.hpp"
+#include <protozero/varint.hpp>
+#include <iostream>
+#include <sstream>
+#include "mvt.hpp"
+#include "write_json.hpp"
+#include "version.hpp"
+
+pmtilesv3 *pmtilesv3_open(const char *filename, char **argv, int force, const char* tmpdir) {
+	pmtilesv3 *outfile = new pmtilesv3;
+
+	struct stat st;
+	if (force) {
+		unlink(filename);
+	} else {
+		if (stat(filename, &st) == 0) {
+			fprintf(stderr, "%s: %s: file exists\n", argv[0], filename);
+			exit(EXIT_FAILURE);
+		}
+	}
+	outfile->ostream.open(filename,std::ios::out | std::ios::binary);
+
+	char tmpname[strlen(tmpdir) + strlen("/pmtiles.XXXXXX") + 1];
+	sprintf(tmpname, "%s%s", tmpdir, "/pmtiles.XXXXXX");
+	int tmpfd = mkstemp(tmpname);
+	close(tmpfd);
+
+	outfile->tmptilesname = tmpname;
+	outfile->tilestmp.open(tmpname,std::ios::out | std::ios::binary);
```

Comment:
access to this should be synchronized via `db_lock`, I don't believe we need anything else threading-related here?